tests/TestCase/View/StringTemplateTest.php

```diff
@@ -300,4 +300,97 @@ public function testPushPopTemplates()
         $this->assertNull($this->template->pop());
         $this->assertNull($this->template->pop());
     }
+
+    public function testAddClassMethod()
+    {
+        // Test new class as null, string, array, false etc
```

Comment:
If you make separate test methods per "scenario", you will not need the inline comments.

testAddClassMethodNewClass
testAddClassMethodCurrentClass

etc for example